controllers/vaultdynamicsecret_controller.go

```diff
@@ -790,7 +790,14 @@ func (r *VaultDynamicSecretReconciler) vaultClientCallback(ctx context.Context,
 
 	reqs := map[client.ObjectKey]empty{}
 	for _, o := range l.Items {
-		if o.Status.VaultClientMeta.CacheKey == cacheKey.String() {
+		if o.Status.VaultClientMeta.CacheKey == "" {
+			logger.V(consts.LogLevelWarning).Info("Skipping, cacheKey is empty",
+				client.ObjectKeyFromObject(&o))
```

Comment:
```suggestion
			logger.V(consts.LogLevelWarning).Info("Skipping, cacheKey is empty",
				"object", client.ObjectKeyFromObject(&o))
```

Reply:
Fixed in 53c5d9fc55f47fd9f9acdc113957617ddaaa4b12